### PR TITLE
Delete all doesn't soft delete

### PR DESCRIPTION
**Pull request by RobinClowers**
I'm not sure if this is expected behavior, but it certainly caught me off guard. I started to look into how to fix it, but I'm not that familiar with AR.

### CODE REVIEW

File: test/paranoia_test.rb [hunk 1]
```diff
@@ -99,7 +99,7 @@ def test_delete_behavior_for_plain_models_callbacks
     assert_equal nil, model.instance_variable_get(:@validate_called)
     assert_equal nil, model.instance_variable_get(:@destroy_callback_called)
     assert_equal nil, model.instance_variable_get(:@after_destroy_callback_called)
-    assert_equal nil, model.instance_variable_get(:@after_commit_callback_called)
+    # assert_equal nil, model.instance_variable_get(:@after_commit_callback_called)
```
Comment (radar) [inline test/paranoia_test.rb]: I would rather this line not be commented out... Either remove it (with an explanation of why it was removed) or keep it.

Comment (RobinClowers) [inline test/paranoia_test.rb]: I commented it out because it was failing before I started. It seemed like it should pass, so I didn't want to remove it.

Comment (radar) [inline test/paranoia_test.rb]: Ok. I would rather it be left failing if it was failing before hand. It's likely that a change in Rails made this happen, and so if that's the case then a failing test indicating that Something's Not Right(tm) is better than wilful ignorance.

Comment (RobinClowers) [inline test/paranoia_test.rb]: Good point, I just rebased that commit out.

**Comment by jhawthorn** [comment]
This is expected behaviour, or at least it's behaviour which would be problematic to change.

We could do a better job of documenting which methods works as soft-deletes. I've opened #304

**Pull request closed without merging**